docs/user-guide/auto_sync.md

```diff
@@ -45,7 +46,10 @@ spec:
   Synced or error state will not attempt automated sync.
 * Automated sync will only attempt one synchronization per unique combination of commit SHA1 and
   application parameters. If the most recent successful sync in the history was already performed
-  against the same commit-SHA and parameters, a second sync will not be attempted.
+  against the same commit-SHA and parameters, a second sync will not be attempted, unless `selfHeal` flag is set to true.
+* If `selfHeal` flag is set to true then sync will be attempted again after self heal timeout (5 seconds by default)
```

Comment:
Is this the best way to configure this? What about using an `argocd-cm` setting?

Reply:
All controllers timeouts are controlled using flags right now. I'm not 100% sure why exactly we split settings this way. I guess because these are only controller related settings and we don't expect end-user to change it at all, except rare cases.

Reply:
@jessesuen ?